### PR TITLE
prevents ai from moving after being killed

### PR DESCRIPTION
**Pull request by AndrewL97**
I don't like that they move, when you kill them

## About The Pull Request

AI used to move when they were killed

https://user-images.githubusercontent.com/22902112/146538770-4db4f624-a4ef-4a56-9598-ad599ace7212.mp4

## Why It's Good For The Game

I used to get really uncomfortable when the stickmen would keep following me 

Now they don't, which is really good

https://user-images.githubusercontent.com/22902112/146538914-87584f67-cec0-4291-9fe9-10006b11897c.mp4

## Changelog

:cl:
fix:ai movement checks for dead or incap
server: MS13 is the best
/:cl:

### CODE REVIEW

**Comment by JohnFulpWillard** [comment]
fix:Do I get more GBPs if I have more fix tags?

No you don't, you get GBP from the label with the most positive gbp gain - the label with the most gbp loss (if you have several)

**Comment by AndrewL97** [comment]
Also to whomestever reviews this - let me know if this is too hacky. It felt gross duplicating the code, but the main ai_controller and movement controllers seem to run /process asynchrnoghusly - so I needed the check in both

File: code/datums/ai/movement/ai_movement_basic_avoidance.dm [hunk 1]
```diff
@@ -6,12 +6,19 @@
 ///Put your movement behavior in here!
 /datum/ai_movement/basic_avoidance/process(delta_time)
 	for(var/datum/ai_controller/controller as anything in moving_controllers)
+
+		var/atom/movable/movable_pawn = controller.pawn
+
+		// Check if this controller can actually run, so we don't chase people with corpses
+		if(!controller.able_to_run())
```
Comment (Fikou) [inline code/datums/ai/movement/ai_movement_basic_avoidance.dm:13]: shouldnt this be after the cooldown code?

**Comment by SuperNovaa41** [comment]
remove all that crap from your changelog its the github tags that give you GBP

File: code/datums/ai/movement/ai_movement_basic_avoidance.dm (hunk 1 above)
Comment (AndrewL97) [reply code/datums/ai/movement/ai_movement_basic_avoidance.dm:13]: I can shift it - I wasn't sure if a hard stop to movement outside of the cooldowns would be better

**Comment by NanoCats** [comment]
That's a bummer, On one hand, yes this is obviously an error, on the other, it was amusing. it'd be cool if "dead" AI could still slowly crawl towards you "in their last breath" for a little and THEN stop. 

Oh well.

File: code/datums/ai/movement/ai_movement_dumb.dm [hunk 2]
```diff
@@ -5,12 +5,20 @@
 ///Put your movement behavior in here!
 /datum/ai_movement/dumb/process(delta_time)
 	for(var/datum/ai_controller/controller as anything in moving_controllers)
+
```
Comment (phasenoisepon) [inline code/datums/ai/movement/ai_movement_dumb.dm:8]: random whitespace?

File: code/datums/ai/movement/ai_movement_basic_avoidance.dm [hunk 3]
```diff
@@ -6,12 +6,19 @@
 ///Put your movement behavior in here!
 /datum/ai_movement/basic_avoidance/process(delta_time)
 	for(var/datum/ai_controller/controller as anything in moving_controllers)
+
```
Comment (phasenoisepon) [inline code/datums/ai/movement/ai_movement_basic_avoidance.dm:9]: random whitespace?

**Comment by AndrewL97** [comment]
@phasenoisepon @tralezab  @Fikou 

Can I please get a re-review? xo

**Comment by iprice** [comment]
4 January 2022
AndrewL97 changed:
 MS13 is the best 

Sure glad I had to go look this up to decode the change log.

**Comment by JohnFulpWillard** [comment]
changelog SABOTAGED by DERANGED downstream coder

**Pull request merged**